src/sinks/datadog/logs.rs

```diff
@@ -2,59 +2,179 @@ use crate::{
+        match &self.endpoint {
+            Some(endpoint) => endpoint,
+            None => "https://http-intake.logs.datadoghq.eu/v1/input",
+        }
+    }
+
+    fn batch_settings<T: Batch>(&self) -> Result<BatchSettings<T>, BatchError> {
+        BatchSettings::default()
+            .bytes(bytesize::kib(100u64))
+            .timeout(1)
+            .parse_config(self.batch)
+    }
+
+    /// Builds the required BatchedHttpSink.
+    /// Since the DataDog sink can create one of two different sinks, this
+    /// extracts most of the shared functionality required to create either sink.
+    fn build_sink<T, B, O>(
+        &self,
+        cx: SinkContext,
+        service: T,
+        batch: B,
+        timeout: Duration,
+    ) -> crate::Result<(VectorSink, Healthcheck)>
+    where
+        O: 'static,
+        B: Batch<Output = Vec<O>> + std::marker::Send + 'static,
+        B::Output: std::marker::Send + Clone,
+        B::Input: std::marker::Send,
+        T: HttpSink<Input = B::Input, Output = B::Output> + Clone,
+    {
+        let request_settings = self.request.unwrap_with(&TowerRequestConfig::default());
 
         let tls_settings = MaybeTlsSettings::from_config(
             &Some(self.tls.clone().unwrap_or_else(TlsConfig::enabled)),
             false,
         )?;
 
-        let sink = TcpSink::new(host, port, cx.resolver(), tls_settings);
-        let healthcheck = sink.healthcheck();
+        let client = HttpClient::new(cx.resolver(), tls_settings)?;
+        let healthcheck = healthcheck(service.clone(), client.clone()).boxed();
+        let sink = BatchedHttpSink::new(
+            service,
+            batch,
+            request_settings,
+            timeout,
+            client,
+            cx.acker(),
+        )
+        .sink_map_err(|e| error!("Fatal datadog_logs text sink error: {}", e));
+
+        Ok((VectorSink::Futures01Sink(Box::new(sink)), healthcheck))
+    }
 
-        let encoding = self.encoding.clone();
-        let api_key = self.api_key.clone();
+    /// Build the request, GZipping the contents if the config specifies.
+    fn build_request(&self, body: Vec<u8>) -> crate::Result<http::Request<Vec<u8>>> {
+        let uri = self.get_endpoint();
+        let request = Request::post(uri)
+            .header("Content-Type", "text/plain")
+            .header("DD-API-KEY", self.api_key.clone());
 
-        let sink = StreamSinkOld::new(sink, cx.acker())
-            .with_flat_map(move |e| iter_ok(encode_event(e, &api_key, &encoding)));
+        let (request, body) = match self.compression {
+            Compression::None => (request, body),
+            Compression::Gzip => {
+                let mut encoder = GzEncoder::new(
+                    Vec::new(),
+                    match self.compression_level {
+                        Some(level) if level <= 9 => flate2::Compression::new(level),
+                        _ => flate2::Compression::fast(),
+                    },
```

Comment:
Yes, I can see that being confusing. I just chose to default to the same level that was being run by the Http sink should an invalid level be entered.

Do you think it might be better to error here instead as if you specify a compression level of 10, you have probably made a mistake?

Reply:
I have gone with just changing this to best. If we do want to validate the parameter, this will need to be done elsewhere anyway, so won't affect this code.

Reply:
Specifying an invalid compression level is probably worth a warning at least, and yes, validation would best be done elsewhere.

Reply:
Ok, I have moved the Compression Level into a newtype. I think it encapsulates the property better, although there is a fair amount of extra boiler plate. 

If you think it is a bit overkill, let me know and I can just put a quick check at the start of the `build` method  instead?

Reply:
It's a little extra boilerplate, but I don't think it's overkill. This will also be useful with support for compression levels in the generic buffer code, so it may end up used elsewhere as well.